### PR TITLE
Add caching page

### PR DESCRIPTION
**Pull request by cb372**

### CODE REVIEW

File: source/caching.md.erb [hunk 1]
```diff
@@ -3,4 +3,81 @@ title: Caching
 nav_order: 50
 ---
 
-TODO explain about caching (copy and translate from wiki)
+Octoparts caches the following information in <%= link_to 'Memcached', 'http://memcached.org/' %>:
```
Comment (lloydmeta) [inline source/caching.md.erb]: I think we should use the simpler and more neutral Markdown syntax for simple links like this, i.e. `[Memcached](http://memcached.org/)`

File: config.rb [hunk 2]
```diff
@@ -41,11 +41,21 @@
...
 # Methods defined in the helpers block are available in templates
-# helpers do
-#   def some_helper
-#     "Helping"
-#   end
-# end
+helpers do
+
+  # Insert a Bootstrap panel with a 'Note' header
+  def note(&block)
```
Comment (lloydmeta) [inline config.rb:47]: nice!

File: source/caching.md.erb (hunk 1 above)
Comment (cb372) [inline source/caching.md.erb]: done

File: source/caching.md.erb [hunk 3]
```diff
@@ -3,4 +3,81 @@ title: Caching
...
+
+For example, say we have three API endpoints, all of which are members of the `userActivityPoints` cache group:
+
+* `getUserProfile`
+* `getUserActivityPointCount`
+* `getUserPremiumStatus`
+
+When user 123's activity point count changes, just make one request to 
+
+    /cache/invalidate/cacheGroup/userActivityPoints/userId/123
```
Comment (lloydmeta) [inline source/caching.md.erb]: I think this is `/cache/invalidate/cacheGroup/userActivityPoints/params/123`, meaning any endpoints with one or more params registered to `userActivityPoints` with a value of 123 will get that portion of their cache invalidated.

Kind of complicated but I think this happens:
1. Assume `getUserProfile` has its `userId` param registered to the `userActivityPoints` cache group.
2. Assume `getUserActivityPointCount` has its `uzerId` (note the z) param registered to the same group
3. `/cache/invalidate/cacheGroup/userActivityPoints/params/123` happens

the result of 3 will be that `getUserProfile`'s version counter for userId:123 will get incremented and `getUserActivityPointCount`'s version counter for uzerId:123 will get incremented.

Comment (cb372) [inline source/caching.md.erb]: Ooh ok, I finally understand how this works now. I'll have to check that I've ported the admin UI for this correctly.

**Pull request merged**